src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,92 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  gasPrice: BigNumber;
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+};
+
+interface GasPriceFeed {
+  (provider: providers.Provider): Promise<GasPriceEstimate>;
+}
+
+async function getProviderFeeData(provider: providers.Provider): Promise<ValidatedFeeData> {
+  const validateFeeData = (val: BigNumber | null): BigNumber  => {
```

Comment:
Nit: why not pull this out of the function so it's not re-initialized on each call?